gapis/api/transform2/transform_chain.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transform2
+
+import (
+	"context"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/commandGenerator"
+	"github.com/google/gapid/gapis/config"
+)
+
+// TransformChain is responsible for running all the transforms on
+// all the commands produced by command enerator
+type TransformChain struct {
+	transforms       []Transform
+	out              Writer
```

Comment:
Should this field be named `writer` ? ("out" is super-generic)

Reply:
This is the old convention that I did not wanted to touch. It conveys the idea of this is the place that where the output of transforms should be processed but let me know if you think it is important to change it.

Reply:
ok